doc/source/topologies.rst

```diff
@@ -0,0 +1,137 @@
+There can be any number of topology files for your project in this directory.
+
+* topologies/my-topology.clj
+* topologies/my-other-topology.clj
+* topologies/my-third-topology.clj
+
+So on and so forth.
+
+Your topology file that describes the Directed Acyclic Graph (DAG) that will 
+be run in Storm.
+
+Looking at ``my-topology.clj``, we start off with importing the streamparse
+Clojure DSL functions
+
+.. code-block:: clojure
+
+    (ns my-topology
+      (:use     [streamparse.specs])
+      (:gen-class))
+
+Notice the ``my-topology`` matches the name of the file. We then have the
+import function.
+
+You could optionally avoid all of the streamparse-provided helper functions and 
+import your own functions or the Clojure DSL for Storm directly. 
+
+.. code-block:: clojure
+
+    (ns my-topology
+      (:use     [backtype.storm.clojure])
+      (:gen-class))
+
+In the next part of the file, we setup a topology definition, also named
+``my-topology`` (matching the ``ns`` line and filename). This definition is 
+actually a Clojure function that takes the topology options as a single map 
+argument. This function returns a list of 2 maps -- a spout map, and a bolt map.
+These two maps define the DAG that is your topology.
+
+.. code-block:: clojure
+
+    (defn my-topology [options]
+       [
+        ;; spout configuration
+        {"my-spout" (python-spout-spec
+              ;; topology options passed in
+              options
+              ;; python class to run
+              "spouts.myspout.MySpout"
+              ;; output specification, what named fields will this spout emit?
+              ["data"]
+              ;; configuration parameters, can specify multiple or none at all
+              )
+        }
+        ;; bolt configuration
+        {"my-bolt" (python-bolt-spec
+              ;; topology options pased in
+              options
+              ;; inputs, where does this bolt receive its tuples from?
+              {"my-spout" :shuffle}
+              ;; python class to run
+              "bolts.mybolt.MyBolt"
+              ;; output specification, what named fields will this spout emit?
+              ["data" "date"]
+              ;; configuration parameters, can specify multiple or none at all
+              :p 2
+              )
+        }
+      ]
+    )
+
+
+Python Spouts and Bolts
+-----------------------
+
+The example topology above, and the ``sparse quickstart wordcount`` project 
+utilizes the ``python-spout-spec`` and ``python-bolt-spec`` provided by the 
+``streamparse.specs`` import statement.
+
+The ``python-spout-spec`` takes at least 3 arguments:
+
+1. ``options`` - the topology options array passed in
+2. The full path to the class to run. ``spouts.myspout.MySpout`` is actually the ``MySpout`` class in ``src/spouts/myspout.py``
+3. A list of the named fields the spout will output
+4. Any optional configuration parameters, such as parallelism
```

Comment:
Should state how these optional configuration parameters are passed. In this case, we pass through everything to ``shell-spout-spec`` and ``shell-bolt-spec``.